File: Sort/quick_sort.py
def partition(arr, l, h):
    pivot = arr[l]
    i = l
    j = h
    while (i < j):
        while pivot >= arr[i]:
            i += 1
        while pivot < arr[j]:
            j -= 1
        if i < j:
            arr[i], arr[j] = arr[j], arr[i]

    arr[j], arr[l] = arr[l], arr[j]
    return j


def quick_sort(arr, l, h):
    if l < h:
        pivot = partition(arr, l, h)
        quick_sort(arr, l, pivot-1)
        quick_sort(arr, pivot+1, h)
    return arr


if __name__ == '__main__':
    arr = [5, 4, 7,3, 2, 10, 7, 8, 9, 1, 5]
    arr1= quick_sort(arr, 0, len(arr)-1)
    print(arr1)

File: algorithm/quick_sort.cpp
#include <iostream>
using namespace std;

int partition(int arr[],int l,int h)
{
    int pivot = arr[h];
    int i = l-1;
    for(int j = l;j<h;j++)
    {
        if(arr[j]<=pivot)
        {
            i++;
            swap(arr[i],arr[j]);
        }
    }
    swap(arr[i+1],arr[h]);
    return(i+1);
}

void quick_sort(int arr[],int l,int h)
{
    if(l<h)
    {
        int p = partition(arr,l,h);
        quick_sort(arr,l,p-1);
        quick_sort(arr,p+1,h);
    }
}

int main()
{
    int arr[] = {1, 3, 8, 9, 4, 3, 2, 1};
    int n = sizeof(arr) / sizeof(arr[0]);
    quick_sort(arr,0,n-1);
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
}